### PR TITLE
bring forward to new graphite 1.1 storage finder interface

### PR DESCRIPTION
**Pull request by rileyberton**
fixes circonus/snowth/#1043

### CODE REVIEW

File: irondb/irondb.py [hunk 1]
```diff
@@ -88,8 +90,9 @@ def fetch(self, start_time, end_time):
                 for i in range(0, min(urls.host_count, tries)):
                     try:
                         self.fetched = False
-                        d = requests.post(urls.series_multi, json = params, headers = self.headers, timeout=((self.connection_timeout / 1000), (self.timeout / 1000)))
-                        if d.headers['content-type'] == 'application/json':
+                        d = requests.post(urls.series_multi, json = params, headers = self.headers,
+                                          timeout=((self.connection_timeout / 1000), (self.timeout / 1000)))
+                        if not 'content-type' in d.headers or d.headers['content-type'] == 'application/json':
```
Comment (cody271) [inline irondb/irondb.py]: If the `content-type` header is missing, isn't that an error?? Should always be present AFAIK.

File: irondb/irondb.py [hunk 2]
```diff
@@ -208,14 +213,86 @@ def __init__(self, config=None):
...
+                    if r.headers['content-type'] == 'application/json':
+                        names = r.json()
+                    elif r.headers['content-type'] == 'application/x-flatbuffer-metric-find-result-list':
+                        names = irondb_flatbuf.metric_find_results(r.content)
+                    else:
+                        pass
+                    break
+                except requests.exceptions.ConnectionError:
+                    # on down nodes, try again on another node until "tries"
+                    pass
```
Comment (cody271) [inline irondb/irondb.py]: Should add the same debug statements from `find_nodes` here I think.

File: irondb/irondb.py [hunk 3]
```diff
@@ -225,17 +302,18 @@ def find_nodes(self, query):
...
             except requests.exceptions.ConnectTimeout:
                 # on down nodes, try again on another node until "tries"
-                pass
+                log.debug("IRONdbFinder.find_nodes ConnectTimeout")
             except requests.exceptions.ReadTimeout:
                 # up node that simply timed out is a failure
+                log.debug("IRONdbFinder.find_nodes ReadTimeout")
                 break
-
+        log.debug("IRONdbFinder.find_nodes, result: %s" % json.dumps(names))
```
Comment (cody271) [inline irondb/irondb.py]: The JSON serialization of the result to the logs can be very expensive, and runs regardless of whether it is printed or not here. This should only happen if the log level is set to debug.

File: irondb/irondb.py (hunk 1 above)
Comment (rileyberton) [reply irondb/irondb.py]: Rearranged this to make it more clear

**Pull request merged**